File: lib/active_admin_bootstrap/extensions/inputs/datetime_picker_input.rb
module ActiveAdminBootstrap
  module Extensions
    module Inputs
      module DatetimePickerInput
        include Base
        include Base::Stringish
        prepend Base::Groupish

        def input_html_options
          super.merge(flatpickr_options)
        end

        def flatpickr_options
          {data: {"aa-datepicker": flatpickr}}
        end

        private

        def flatpickr
          options[:flatpickr] || inputs_config.dig(:datetime_picker)
        end
      end
    end
  end
end

module ActiveAdmin
  module Inputs
    class DatetimePickerInput < ::Formtastic::Inputs::DatetimePickerInput
      include ::ActiveAdminBootstrap::Extensions::Inputs::DatetimePickerInput
    end
  end
end

File: app/javascript/active_admin_bootstrap/lib/inputs/datetime_picker_input.js
import BaseInput from './base_input'

class DateTimePickerInput extends BaseInput {
  _defaultInputHTMLOptions () {
    const options = {
      type: 'text',
      'data-aa-datepicker': {
        enableTime: true,
        altFormat: 'Y-m-d, h:i K'
      }
    }

    return { ...super._defaultInputHTMLOptions(), ...options }
  }
}

export default DateTimePickerInput
